internal/handlers/pg/msg_getparameter.go

```diff
@@ -25,25 +26,82 @@ import (
 
 // MsgGetParameter OpMsg used to get parameter.
 func (h *Handler) MsgGetParameter(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	document, err := msg.Document()
+	cmd, err := msg.Document()
 	if err != nil {
 		return nil, lazyerrors.Error(err)
 	}
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/449
+	getPrm, err := cmd.Get("getParameter")
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
 
-	common.Ignored(document, h.l, "comment")
+	// SELECT * FROM 'admin'
+	resDB := types.MustNewDocument(
+		"acceptApiVersion2", false,
+		"authSchemaVersion", int32(5),
+		"quiet", false,
+		"ok", float64(1),
+	)
 
 	var reply wire.OpMsg
+	var errNoFound error
+	var resDoc *types.Document
+	if getPrm == "*" {
+		resDoc = resDB
+	} else {
+		resDoc, errNoFound, err = selectParam(cmd, resDB)
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+	}
+
 	err = reply.SetSections(wire.OpMsgSection{
-		Documents: []*types.Document{types.MustNewDocument(
-			"quiet", false,
-			"ok", float64(1),
-		)},
+		Documents: []*types.Document{resDoc},
 	})
 	if err != nil {
 		return nil, lazyerrors.Error(err)
 	}
 
-	return &reply, nil
+	comment, err := cmd.Get("comment")
+	if err == nil {
+		common.Ignored(cmd, h.l, fmt.Sprint(comment))
+	}
+
+	return &reply, errNoFound
+}
+
+// selectParam is makes a selection of the requested parameters received from database 'admin'.
+func selectParam(cmd, resDB *types.Document) (doc *types.Document, errNoFound error, err error) {
+	doc = types.MustNewDocument()
+	keys := cmd.Keys()
+
+	for _, k := range keys {
+		if k == "getParameter" || k == "comment" || k == "$db" {
+			continue
+		}
+		item, err := resDB.Get(k)
+		if err != nil {
+			continue
+		}
+		err = doc.Set(k, item)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	if doc.Len() < 1 {
+		err := doc.Set("ok", float64(0))
+		if err != nil {
+			return nil, nil, err
+		}
+		errNoFound = common.NewErrorMsg(common.ErrorCode(0), "no option found to get")
```

Comment:
here you can make an early return and the other part of code will become more readable

Reply:
fix